bundles/org.openhab.binding.flume/src/main/java/org/openhab/binding/flume/internal/handler/FlumeBridgeHandler.java

```diff
@@ -0,0 +1,299 @@
+
+    private final FlumeApi api;
+    final SystemOfUnits systemOfUnits;
+    final TranslationProvider i18nProvider;
+    final LocaleProvider localeProvider;
+    final Bundle bundle;
+
+    public FlumeApi getApi() {
+        return api;
+    }
+
+    protected @Nullable ScheduledFuture<?> pollingJob;
+    private @Nullable FlumeDiscoveryService discoveryService;
+
+    /**
+     * Get the services registered for this bridge. Provides the discovery service.
+     */
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(FlumeDiscoveryService.class);
+    }
+
+    public boolean registerDiscoveryListener(FlumeDiscoveryService listener) {
+        if (discoveryService == null) {
+            discoveryService = listener;
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean unregisterDiscoveryListener() {
+        if (discoveryService != null) {
+            discoveryService = null;
+            return true;
+        }
+
+        return false;
+    }
+
+    public FlumeBridgeHandler(final Bridge bridge, SystemOfUnits systemOfUnits, HttpClient httpClient,
+            TranslationProvider i18nProvider, LocaleProvider localeProvider) {
+        super(bridge);
+
+        api = new FlumeApi(httpClient);
+        this.systemOfUnits = systemOfUnits;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(this.getClass());
+    }
+
+    public FlumeBridgeConfig getFlumeBridgeConfig() {
+        return config;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FlumeBridgeConfig.class);
+
+        if (config.clientId.isBlank() | config.clientSecret.isBlank() || config.password.isBlank()
+                || config.username.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "@text/offline.cloud-configuration-error");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(this::goOnline);
+    }
+
+    public void goOnline() {
+        try {
+            api.initialize(config.clientId, config.clientSecret, config.username, config.password,
+                    this.getThing().getUID());
+        } catch (FlumeApiException | IOException | InterruptedException | TimeoutException | ExecutionException e) {
+            handleApiException(e);
+            return;
+        }
+
+        if (!refreshDevices(true)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.cloud-configuration-error");
+            return;
+        }
+
+        int pollingPeriod = Math.min(config.refreshIntervalCumulative, config.refreshIntervalInstantaneous);
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevices, pollingPeriod, pollingPeriod,
```

Comment:
```suggestion
        pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevices, 0, pollingPeriod,
```
Before the last commit there was no initial delay, not sure why you added it here, but i recommend to revert this to zero delay.

Reply:
as part of the device coming online, it makes the same query calls so it was to prevent a double call initially when both the bridge came online and when the device did.  I did change it back since its not doesn't impact the functionality at all.